internal/worker/modelworkermanager/modelworkermanager.go

```diff
@@ -240,36 +238,24 @@ func (m *modelWorkerManager) modelChanged(modelUUID string) error {
 	} else if err != nil {
 		return errors.Trace(err)
 	}
-	defer release()
-
-	if !isModelActive(model) {
-		// Ignore this model until it's activated - we
-		// never want to run workers for an importing
-		// model.
-		// https://bugs.launchpad.net/juju/+bug/1646310
-		return nil
-	}
 
 	cfg := NewModelConfig{
 		Authority:    m.config.Authority,
-		ModelName:    model.Name(),
-		ModelOwner:   model.Owner().Id(),
+		ModelName:    model.Name,
+		ModelOwner:   model.Owner.String(),
```

Comment:
Owner is a UUID
This needs to be OwnerName - the original code has Owner().Id() which is the tag id which is the name
(plus test needs to be updated)

Reply:
Refactored to model.OwnerName.Name(), thanks Ian!